compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -627,6 +618,29 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
       }
     }
 
+    private def reportCannotDisambiguateError(alts: List[Exported]): Unit = {
+      val currentClass = currentClassSym.get
+
+      // Find a position that is in the current class for decent error reporting
+      val pos = alts.collectFirst {
+        case ExportedSymbol(sym) if sym.owner == currentClass => sym.pos
+        case alt: ExportedBody                                => alt.pos
+      }.getOrElse {
+        currentClass.pos
+      }
```

Comment:
Hmmm... What about something like this:

```scala
trait Foo extends js.Object {
  def foo: js.UndefOr[Int] = js.undefind
}

trait Bar extends js.Object {
  @JSName("foo")
  def bar: js.UndefOr[Int] = js.undefined
}

class A extends Foo with Bar
```

Shouldn't this complain (but probably wont at the moment)?